src/main/kotlin/io/sip3/twig/ce/controller/HostController.kt

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2018-2020 SIP3.IO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.sip3.twig.ce.controller
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import io.sip3.twig.ce.domain.Host
+import io.sip3.twig.ce.service.HostService
+import io.swagger.annotations.Api
+import io.swagger.annotations.ApiOperation
+import io.swagger.annotations.ApiResponse
+import io.swagger.annotations.ApiResponses
+import org.springframework.dao.DuplicateKeyException
+import org.springframework.web.bind.annotation.*
+import org.springframework.web.multipart.MultipartFile
+import springfox.documentation.swagger2.annotations.EnableSwagger2
+import javax.validation.ConstraintViolationException
+import javax.validation.Valid
+import javax.validation.Validation.buildDefaultValidatorFactory
+import javax.validation.constraints.NotNull
+
+@EnableSwagger2
+@Api(
+        tags = ["Host Controller"]
+)
+@RestController
+@RequestMapping("/hosts")
+class HostController(private val hostService: HostService, private val mapper: ObjectMapper) {
+
+    @ApiOperation(
+            position = 1,
+            value = "Get all hosts",
+            produces = "application/json"
+    )
+    @ApiResponses(value = [
+        ApiResponse(code = 200, message = "Returns all hosts"),
+        ApiResponse(code = 500, message = "Internal Server Error"),
+        ApiResponse(code = 504, message = "Connection timeout")]
+    )
+    @GetMapping
+    fun getAll(): Set<Host> {
```

Comment:
listHosts - if we follow `AttributeController` naming

Also I like:
1. createHost
2. updateHost
3. getHost (without by name- it's clear)
4. deleteHost

Reply:
Not sure we need to append method with `*Host`. This controller is responsible for hosts management.